pubsub/pstest/fake.go

```diff
@@ -368,8 +368,26 @@ func (s *GServer) UpdateTopic(_ context.Context, req *pb.UpdateTopicRequest) (*p
 				return nil, err
 			}
 			t.proto.MessageRetentionDuration = req.Topic.MessageRetentionDuration
-		case "schema_settings":
-			t.proto.SchemaSettings = req.Topic.SchemaSettings
+		case "schema_settings.schema":
+			if t.proto.SchemaSettings == nil {
+				t.proto.SchemaSettings = &pb.SchemaSettings{}
+			}
+			t.proto.SchemaSettings.Schema = req.Topic.SchemaSettings.Schema
+		case "schema_settings.encoding":
+			if t.proto.SchemaSettings == nil {
+				t.proto.SchemaSettings = &pb.SchemaSettings{}
+			}
+			t.proto.SchemaSettings.Encoding = req.Topic.SchemaSettings.Encoding
+		case "schema_settings.first_revision_id":
+			if t.proto.SchemaSettings == nil {
+				t.proto.SchemaSettings = &pb.SchemaSettings{}
+			}
+			t.proto.SchemaSettings.FirstRevisionId = req.Topic.SchemaSettings.FirstRevisionId
+		case "schema_settings.last_revision_id":
+			if t.proto.SchemaSettings == nil {
+				t.proto.SchemaSettings = &pb.SchemaSettings{}
+			}
+			t.proto.SchemaSettings.LastRevisionId = req.Topic.SchemaSettings.LastRevisionId
 		default:
```

Comment:
It is also possible to just have schema_settings, which will update all of them. Though I guess the Go library is setting each field explicitly?

Reply:
Yeah each field is set explicitly so updating all of them at once is possible too.